--- src/pages/Quizz.tsx
+++ src/pages/Quizz.tsx
@@ -20,7 +20,7 @@
   const [showAnswer, setShowAnswer] = useState(false);
   const [isFinished, setIsFinished] = useState(false);
   const [isPaused, setIsPaused] = useState(false);
-  const [timeSpent, setTimeSpent] = useState(0);
+  const timeSpent = React.useRef(0);
   const [open, setOpen] = React.useState(false);
   const notificationContent = (questionNumber: number, time: string) => (
     <div className="toast-content">
@@ -34,31 +34,25 @@
     return "error";
   };
-  const notifyTime = () => {
-    if (timeSpent <= 3) return;
+
+  const notifyTime = () => {
+    const time = timeSpent.current;
+    if (time <= 3) return;
+
+    const toastOptions: any = {
+      position: "bottom-right",
+      autoClose: 3000,
+      hideProgressBar: true,
+      closeOnClick: false,
+      pauseOnHover: true,
+      theme: "colored",
+      draggable: false,
+      closeButton: false,
+      type: toastType(time),
+    };
+
     toast(
-      notificationContent(currentQuestion, formatTime(timeSpent)),
-      toastOptions
+      notificationContent(currentQuestion, formatTime(time)),
+      toastOptions
     );
   };

-  const toastOptions: any = {
-    position: "bottom-right",
-    autoClose: 3000,
-    hideProgressBar: true,
-    closeOnClick: false,
-    pauseOnHover: true,
-    theme: "colored",
-    draggable: false,
-    closeButton: false,
-    type: toastType(timeSpent),
-  };
-
   useEffect(() => {
-    setTimeSpent(0);
+    timeSpent.current = 0;
   }, [currentQuestion]);

   useEffect(() => {
@@ -66,7 +60,7 @@

     if (!isPaused) {
       intervalId = setInterval(() => {
-        setTimeSpent((prevTime) => prevTime + 1);
+        timeSpent.current += 1;
       }, 1000);
     } else {
       clearInterval(intervalId);
